python/paddle/jit/sot/opcode_translator/executor/variable_dispatch.py

```diff
@@ -981,16 +981,17 @@ def str_endswith(
     ),
 )
 
+
 # VariableBase
-Dispatcher.register(
-    operator.is_,
-    ("VariableBase", "VariableBase"),
-    lambda var, other: ConstantVariable(
-        var.get_py_value() is other.get_py_value(),
+@Dispatcher.register_decorator(operator.is_)
+def is_func(var: VariableBase, other: VariableBase):
+    if var.get_py_type() != other.get_py_type():
```

Comment:
```suggestion
    if var.get_py_type() is not other.get_py_type():
```

这里可以改成 `is not`，不需要触发 `__eq__` 的 dispatch，直接严格相同对象即可

Reply:
好的